xbmc/FileItem.cpp

```diff
@@ -1489,7 +1489,7 @@ void CFileItem::FillInMimeType(bool lookup /*= true*/)
 
   // change protocol to mms for the following mime-type.  Allows us to create proper FileMMS.
   if( StringUtils::StartsWithNoCase(m_mimetype, "application/vnd.ms.wms-hdr.asfv1") || StringUtils::StartsWithNoCase(m_mimetype, "application/x-mms-framed") )
-    StringUtils::Replace(m_strPath, "http:", "mms:");
+    StringUtils::Replace(m_strDynPath, "http:", "mms:");
```

Comment:
same here

Reply:
This is a replace I can only do `!m_strDynPath.empty() ? m_strDynPath : m_strPath` instead.
Anyway we change m_strPath here, which is actually not desired for a function that should just fill a mimetype right?
Not sure how to fix this though...

Reply:
> EDIT: disregard, m_strPath must never be changed. if m_strDynPath is empty it has to be filled with m_strPath before replace

Do we want to do this in here?
I'd prefer setting DynPath when Path is set (of course only if it's empty at this stage), what's your opinion?

Reply:
Only strDynPath can be changed. if dynPath is not empty, do the replace. if dynpath is empty, copy from path and do replace.

Reply:
done